### PR TITLE
Add some comments to domtree code

### PR DESCRIPTION
**Pull request by oxinabox**
This is part of the result of me quizing @Keno on this stuff.

### CODE REVIEW

File: base/compiler/ssair/domtree.jl [hunk 1]
```diff
@@ -302,6 +313,7 @@ begin
             end
             idoms_dfs[v] = idom
         end
+        # The following produces for each BB in cfg, a map to its parent (as indexed in the cfg BB)
```
Comment (Keno) [inline base/compiler/ssair/domtree.jl]: Probably not incorrect, but really all this does is translate the idom relationship from dfs indexing to bb# indexing.

Comment (oxinabox) [reply base/compiler/ssair/domtree.jl]: Ah yes, so it is. I thought (but didn't write)
that this was also moving up 1 level, but `idoms` always was referring to up one level.

```suggestion
        # Reexpress the idom relationship in  BB indexing
```

File: base/compiler/ssair/domtree.jl [hunk 2]
```diff
@@ -1,18 +1,27 @@
...
     idoms::Vector{Int}
+    
+    # The nodes in the tree (ordered by BB indicies)
     nodes::Vector{DomTreeNode}
 end
 
 """
-    Checks if bb1 dominates bb2
+    Checks if bb1 dominates bb2.
+    bb1 and bb2 are indexes into the
```
Comment (Keno) [inline base/compiler/ssair/domtree.jl]: missing sentence?

Comment (oxinabox) [reply base/compiler/ssair/domtree.jl]:
```suggestion
    bb1 and bb2 are indexes into the CFG blocks.
    bb1 dominates bb2 if the only way to enter bb2 is via bb1. 
    (Other blocks may be in between, e.g bb1->bbX->bb2).
```

File: base/compiler/ssair/domtree.jl [hunk 3]
```diff
@@ -1,18 +1,29 @@
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
+# Represents a Basic Block, in the DomTree
 struct DomTreeNode
+    # How deep we are in the DomTree
     level::Int
+    # The BB indicies in the CFG for all Basic Blocks we immidately dominate
```
Comment (jekbradbury) [inline base/compiler/ssair/domtree.jl]: s/indicies/indices
s/immidately/immediately
(both also show up one more time)

File: base/compiler/ssair/domtree.jl [hunk 4]
```diff
@@ -38,11 +49,13 @@ function update_level!(domtree::Vector{DomTreeNode}, node::Int, level::Int)
     end
 end
 
+"Iterable data structure that walks though all dominated blocks"
 struct DominatedBlocks
     domtree::DomTree
     worklist::Vector{Int}
 end
 
+"Returns an interator that walks through all blocks dominated by the basic block at index `root`"
```
Comment (vchuravy) [inline base/compiler/ssair/domtree.jl]:
```suggestion
"Returns an iterator that walks through all blocks dominated by the basic block at index `root`"
```

File: base/compiler/ssair/domtree.jl [hunk 5]
```diff
@@ -241,6 +254,10 @@ begin
     end
 
     """
+        SNCA(cfg::CFG)
```
Comment (vchuravy) [inline base/compiler/ssair/domtree.jl]:
```suggestion
        SNCA(cfg::CFG)

```

File: base/compiler/ssair/domtree.jl [hunk 6]
```diff
@@ -302,6 +319,7 @@ begin
             end
             idoms_dfs[v] = idom
         end
+        # Reexpress the idom relationship in  BB indexing
```
Comment (vchuravy) [inline base/compiler/ssair/domtree.jl]:
```suggestion
        # Reexpress the idom relationship in BB indexing
```

**Comment by oxinabox** [comment]
@Keno  can we merge this?

**Pull request merged**